python/dlr/counter/system.py

```diff
@@ -0,0 +1,109 @@
+from .utils.helper import *
+
+# Interface
+class System:
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def get_device_info(self):
+        """Return a list of device information"""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_device_uuid(self):
+        """Return DeviceInfo uuid"""
+        raise NotImplementedError
+
+
+# Wrapper class
+class ARM(System, ABC):
+    pass
+
+
+class Linux_ARM(ARM):
+    def __init__(self):
+        self._device = DeviceInfo()
+
+        try:
+            # retrieve device information
+            self._device.machine = platform.machine()
+            self._device.arch = platform.architecture()[0]
+            _uuid = ':'.join(
+                ['{:02x}'.format((uuid.getnode() >> ele) & 0xff) for ele in range(0, 8 * 6, 8)][::-1])
+            _md5uuid = get_hash_string(_uuid.encode()) 
+            self._device.uuid = str(_md5uuid.hexdigest())
+            self._device.osname = platform.system()
+            dist = platform.dist()
+            self._device.dist = " ".join(x for x in dist)
+            self._device.name = platform.node()
+        except Exception as e:
+            logging.exception("linux_arm api exception occurred", exc_info=True)
+
+    def get_device_info(self):
+        """Return a list of fields of device information"""
+        return self._device.get_info()
+
+    def get_device_uuid(self):
+        """Return DeviceInfo uuid"""
+        return self._device.uuid
+
+
+class Android(ARM, ABC):
+    pass
+
+
+class X86(System, ABC):
+    pass
+
+
+class Linux_x86(X86):
+    def __init__(self):
+        self._device = DeviceInfo()
+
+        try:
+            # retrieve device information
+            self._device.machine = platform.machine()
+            self._device.arch = platform.architecture()[0]
+            _uuid = ':'.join(
+                ['{:02x}'.format((uuid.getnode() >> ele) & 0xff) for ele in range(0, 8 * 6, 8)][::-1])
+            _md5uuid = get_hash_string(_uuid.encode()) 
+            self._device.uuid = str(_md5uuid.hexdigest())
+            self._device.osname = platform.system()
+            dist = platform.dist()
+            self._device.dist = " ".join(x for x in dist)
+            self._device.name = platform.node()
+        except Exception as e:
+            logging.exception("linux_x86 api exception occurred", exc_info=True)
+
+    def get_device_info(self):
+        """Return a list of fields of device information"""
+        return self._device.get_info()
+
+    def get_device_uuid(self):
+        """Return DeviceInfo uuid"""
+        return self._device.uuid
+
+
+# mapped system types
+system_list = ["Linux_ARM", "Linux_x86"]
+
+
+# factory class for System wrapper class
+class Factory:
+    @staticmethod
+    def get_system(sys_typ):
+        """Return instance of System as per operating system type"""
+        try:
+            map_sys_typ = [item for item in system_list if item.lower() in sys_typ.lower()]
```

Comment:
@dkasar-aws  I talked to @ziyu-guo. This system support validation is not necessary correct based on this https://github.com/neo-ai/neo-ai-dlr/blob/master/doc/install.rst#table-1-list-of-supported-devices

If I understand correctly, you're trying to separate implementation between Android and Linux, right?

I reached this problem while testing DLR on other linux distributions. Based on my conversation with other device engineers, linux distro shouldn't be a problem. 

If that's the case, please consider add support for all other linux distributions as well.

Reply:
Assume all these devices on Linux OS, distributions. We have tested the feature on Linux 86, and ARM architecture machines.
Let us know to expand scope of the feature.

Reply:
We can expand this for all machines for linux. Phone-home shouldn't be architecture dependent. (at least not in python proportion)